### PR TITLE
Add Cloud IDS Pack 2

### PR DESCRIPTION
**Pull request by miamiasheep**
## Contributing to Cortex XSOAR Content
This is the project under the sale's team of Palo Alto Networks.
We create a new content pack called CloudIDS. It now contains a playbook that update ip blocklist to the GCE firewall

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CloudIDS/Playbooks/IPBlacklistGCPFirewall.yml [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+    skipunavailable: false
+    quietmode: 0
+  "37":
+    id: "37"
+    taskid: 49934a26-f7ac-4798-8779-2850008b3085
+    type: title
+    task:
+      id: 49934a26-f7ac-4798-8779-2850008b3085
+      version: -1
+      name: GCP Firewall Name Not Founnd
```
Comment (yaakovi) [inline Packs/CloudIDS/Playbooks/IPBlacklistGCPFirewall.yml:364]:
```suggestion
      name: GCP Firewall Name Not Found
```

File: Packs/CloudIDS/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# CloudIDS
+Google Cloud IDS, a next-generation advanced intrusion detection service that provides threat detection for intrusions, malware, spyware and command-and-control attacks.
```
Comment (yaakovi) [inline Packs/CloudIDS/README.md:2]:
```suggestion
Google Cloud IDS, a next-generation advanced intrusion detection service that provides threat detection for intrusions, malware, spyware, and command-and-control attacks.

```

File: Packs/CloudIDS/IncidentTypes/CloudIDS.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
+[
+	{
+		"id": "Cloud IDS",
```
Comment (evisochek) [inline Packs/CloudIDS/IncidentTypes/CloudIDS.json:3]: Please change to google cloud IDS

File: Packs/CloudIDS/IncidentTypes/CloudIDS.json [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+		"packID": "",
+		"itemVersion": "",
+		"fromServerVersion": "",
+		"toServerVersion": "",
+		"vcShouldIgnore": false,
+		"vcShouldKeepItemLegacyProdMachine": false,
+		"commitMessage": "",
+		"shouldCommit": false,
+		"locked": false,
+		"name": "Cloud IDS",
```
Comment (evisochek) [inline Packs/CloudIDS/IncidentTypes/CloudIDS.json:16]: Please change to google cloud IDS

File: Packs/CloudIDS/IncidentTypes/CloudIDS.json [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+		"prevName": "Cloud IDS",
+		"color": "#F8E7A5",
+		"playbookId": "IP Blacklist - GCP Firewall",
```
Comment (evisochek) [inline Packs/CloudIDS/IncidentTypes/CloudIDS.json:19]: Please change to: "GCP Firewall - Blacklist IP"

File: Packs/CloudIDS/IncidentTypes/CloudIDS.json (hunk 3 above)
Comment (miamiasheep) [reply Packs/CloudIDS/IncidentTypes/CloudIDS.json]: Fixed Accordingly

File: Packs/CloudIDS/IncidentTypes/CloudIDS.json (hunk 4 above)
Comment (miamiasheep) [reply Packs/CloudIDS/IncidentTypes/CloudIDS.json]: Fixed accordingly

File: Packs/CloudIDS/IncidentTypes/CloudIDS.json (hunk 5 above)
Comment (miamiasheep) [reply Packs/CloudIDS/IncidentTypes/CloudIDS.json]: Change Accordingly

File: Packs/CloudIDS/README.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# CloudIDS
+Google Cloud IDS, a next-generation advanced intrusion detection service that provides threat detection for intrusions, malware, spyware, and command-and-control attacks.
```
Comment (mchasepan) [inline Packs/CloudIDS/README.md:2]:
```suggestion
Google Cloud IDS is a next-generation advanced intrusion detection service that provides threat detection for intrusions, malware, spyware, and command-and-control attacks.

```

File: Packs/CloudIDS/README.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
+# CloudIDS
+Google Cloud IDS, a next-generation advanced intrusion detection service that provides threat detection for intrusions, malware, spyware, and command-and-control attacks.
+
+
+
+## What does this packs do?
```
Comment (mchasepan) [inline Packs/CloudIDS/README.md:6]:
```suggestion
## What does this pack do?

```

File: Packs/CloudIDS/README.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
+# CloudIDS
+Google Cloud IDS, a next-generation advanced intrusion detection service that provides threat detection for intrusions, malware, spyware, and command-and-control attacks.
+
+
+
+## What does this packs do?
+### Playbook
+* `IP Blacklist - GCP Firewall`: Gets the attacker's IP address from
+  Cloud IDS through Google Pub/Sub. Then, `IP Blacklist - GCP Firewall` will update the ip list to patch GCP
```
Comment (mchasepan) [inline Packs/CloudIDS/README.md:9]:
```suggestion
  Cloud IDS through Google Pub/Sub. Then `IP Blacklist - GCP Firewall` will update the ip list so GCP automatically blocks the IP address

```

**Comment by evisochek** [comment]
@miamiasheep Pending our session today please do the follow:
divide you playbook into 2:
1) first playbook will fetch the JSON and will extract the SRC IP's
2) second playbook will block the relevant IP's that he will receive as an input from the first playbook.

after creating the playbooks above please wrap them in parent playbook that will contain the relevant incident type.

@EliorKedar

**Comment by evisochek** [comment]
Please create individual readme file file for each playbook and make sure to upload the playbooks images and to add the images links to each playbook readme.
after your modification lets schedule a demo.

**Comment by evisochek** [comment]
@Chaosshade99 Please confirm that all the paths in the playbooks are ending in the Done section header:
image

File: Packs/CloudIDS/Playbooks/Cloud_IDS-IP_Blacklist-GCP_Firewall_Append.yml [hunk 9]
```diff
@@ -0,0 +1,397 @@
...
+    runFromLastJobTime: false
+- key: GCPFirewallName
+  value: {}
+  required: true
+  description: Name of the GCP Firewall where the playbook should set the IPs
+  playbookInputQuery: null
+- key: srcip
+  value: {}
+  required: false
+  description: ""
```
Comment (evisochek) [inline Packs/CloudIDS/Playbooks/Cloud_IDS-IP_Blacklist-GCP_Firewall_Append.yml:393]: Please add description for the input.

File: Packs/CloudIDS/Playbooks/Cloud_IDS-IP_Blacklist-GCP_Firewall_README.md [hunk 10]
```diff
@@ -0,0 +1,11 @@
+# Cloud_IDS-IP_Blacklist-GCP_Firewall
```
Comment (evisochek) [inline Packs/CloudIDS/Playbooks/Cloud_IDS-IP_Blacklist-GCP_Firewall_README.md:1]: I think that you create readme file for each playbook so this readme file could be removed.

**Review by evisochek — APPROVED** [review]

**Comment by yaakovi** [comment]
@Chaosshade99 
please fix the issues in the [build](https://app.circleci.com/pipelines/github/demisto/content/115037/workflows/0ebf996c-a5ce-42b9-b4e8-808369bcc096/jobs/425280/parallel-runs/0/steps/0-109).
they are caused by an invalid structure of your files and can be resolved using the `demisto-sdk format` command.

**Pull request merged**